src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Filters/IdentityTelemetryFilter.cs

```diff
@@ -0,0 +1,56 @@
+using System.Diagnostics.CodeAnalysis;
+
+using Altinn.App.PlatformServices.Extensions;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+
+namespace Altinn.App.PlatformServices.Filters
+{
+    /// <summary>
+    /// Filter to enrich request telemetry with identity information
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class IdentityTelemetryFilter : ITelemetryProcessor
+    {
+        private ITelemetryProcessor Next { get; set; }
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IdentityTelemetryFilter"/> class.
+        /// </summary>
+        public IdentityTelemetryFilter(ITelemetryProcessor next, IHttpContextAccessor httpContextAccessor)
+        {
+            Next = next;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <inheritdoc/>
+        public void Process(ITelemetry item)
+        {
+            RequestTelemetry request = item as RequestTelemetry;
+
+            if (request != null)
+            {
+                HttpContext ctx = _httpContextAccessor.HttpContext;
+
+                if (ctx?.User != null)
+                {
+                    int? orgNumber = ctx.User.GetOrgNumber();
+                    int? userId = ctx.User.GetUserIdAsInt();
+                    int? partyId = ctx.User.GetPartyIdAsInt();
+                    int authLevel = ctx.User.GetAuthenticationLevel();
+
+                    string idString = $"partyId:{partyId};userId:{userId};orgNumber:{orgNumber};authLevel:{authLevel}";
+
+                    item.Context.User.Id = idString;
```

Comment:
Have you considered using the properties dictionary on the RequestTelementry object, adding each of the four values as separate entries instead of concatenating a string into one property? According to the doc this looks like the way to add multiple custom properties: _"Gets a dictionary of application-defined property names and values providing additional information about this request."_

The userId could go into the User.Id, but check out the User.AuthenticatedUserId as well. 

I assume we in the end in application insights will get better query support by adding these as separate properties.

Reply:
We use User.AuthenticatedUserId in the platform for userId/orgNo, but as what this value represents may vary I figured we might leave it out of the app telemetry and only use the named values.
 
We can use the property dictionary instead for the values.